src/components/SwitchButton/index.tsx

```diff
@@ -80,11 +123,31 @@ const SwitchButton = ({
     setLocalValue(event.target.value)
   }
 
+  useEffect(() => {
+    if (!leftButtonRef.current || !rightButtonRef.current) return
+    setLeftCardWidth(leftButtonRef.current.getBoundingClientRect().width)
+    setRightCardWidth(rightButtonRef.current.getBoundingClientRect().width)
+  }, [leftButton.value, leftButtonRef, localValue, rightButtonRef])
+
+  const setMouseDown = (isMouseDown: boolean) => () =>
+    setHasMouseDown(isMouseDown)
+
   return (
     <Tooltip text={tooltip}>
       <div style={{ display: 'inline-flex' }}>
-        <StyledBorderedBox>
+        <StyledBorderedBox
+          onMouseDown={setMouseDown(true)}
+          onMouseUp={setMouseDown(false)}
+          onMouseLeave={setMouseDown(false)}
+        >
+          <FocusOverlay
+            leftCardWidth={leftCardWidth}
+            rightCardWidth={rightCardWidth}
```

Comment:
Can we use data attributes instead?

Reply:
data attributes would only work for discrete values, so can't do this for `width` and `transform` props. But I now compute them in the inline `style` (to avoid changing the class name on every change). I used data attribute for focus position.

Let me know what you think. Can we do better?